sdk/core/azure-core/CLIENT_LIBRARY_DEVELOPER.md

```diff
@@ -601,6 +601,264 @@ manager, with `__aenter__`, `__aexit__`, and `close` methods.
+     https://aka.ms/azsdk/autorest/openapi/lro-options
+    """
+
+    def get_status(self, pipeline_response: "PipelineResponseType") -> str:
+        response = pipeline_response.http_response
+        if _is_empty(response):
+            raise BadResponse("The response from long running operation does not contain a body.")
+
+        body = _as_json(response)
+        status = body.get("status")
+        if not status:
+            raise BadResponse("No status found in body")
+        if status.lower() == "validationfailed":
+            raise HttpResponseError(response=response, error=body["error"])
+        return status
+```
+
+To plug into a client method which was generated as an LRO, you can pass the `polling` keyword argument. The `polling`
+keyword argument takes a `azure.core.polling.PollingMethod` and allows for a custom strategy to be passed in to the keyword
+argument `lro_algorithms`:
+
+```python
+class ServiceOperations:
+
+    def begin_analyze(self, data: AnyStr, name: str) -> LROPoller[JSON]:
+        return self._client.begin_analyze(
+            data,
+            name,
+            polling=LROBasePolling(
+                lro_algorithms=[
+                    CustomOperationResourcePolling()  # overrides other LRO strategies
+                ]
+            ),
+            **kwargs
+        )
+```
+
+This example uses the default polling method - `LROBasePolling` and just overrides the algorithm used for polling.
+If you need to control the polling loop, then see the next section.
+
+
+### Polling method - LROBasePolling/AsyncLROBasePolling
+
+Built-in methods for polling are included in `azure-core` as both sync / async variants - `LROBasePolling` and `AsyncLROBasePolling`.
+You can also use `azure.core.polling.NoPolling` which will not initiate polling and simply return the 
+deserialized initial response when called with `poller.result()`.
+
+To use `NoPolling`, you can simply pass `polling=False` to an operation generated as an LRO:
+
+```python
+class ServiceOperations:
+
+    def begin_analyze(self, data: AnyStr, name: str) -> LROPoller[JSON]:
+        return self._client.begin_analyze(
+            data,
+            name,
+            polling=False,
+            **kwargs
+        )
+```
+
+The polling method runs the polling loop and performs GET requests to the status monitor to check if a terminal state is reached. 
+In between polls it inserts delay based on the service sent `retry-after` header or the given `polling_interval`.
+
+To customize parts of the polling method, you can create a subclass which uses `LROBasePolling` and overrides necessary methods.
+If significant customization is necessary, use `azure.core.polling.PollingMethod` and implement all the necessary methods.
+
+Example: Create an LRO method which will poll for when a file gets uploaded successfully (greatly simplified)
+
+
+```python
+class CustomLROPolling(PollingMethod):
```

Comment:
I think this demo class really needs more documentation. It's difficult to see where the points of customization are that would have needed a Custom class. A description of how each method in the API is used and what it needs to achieve and how those outcomes can be manipulated would be helpful.

Reply:
I added some docstrings / inline comments to explain what should be done in each method. Let me know if you think there's still room for improvement. :)